File: Data_Structures/Queue/Queue.js

// creating object for queue class 
var queue = new Queue(); 
              
  
// Testing dequeue and pop on an empty queue 
// returns Underflow 
console.log(queue.dequeue()); 
  
// returns true 
console.log(queue.isEmpty()); 
  
// Adding elements to the queue 
// queue contains [10, 20, 30, 40, 50] 
queue.enqueue(10); 
queue.enqueue(20); 
queue.enqueue(30); 
queue.enqueue(40); 
queue.enqueue(50); 
queue.enqueue(60); 
  
// returns 10 
console.log(queue.front()); 
  
// removes 10 from the queue 
// queue contains [20, 30, 40, 50, 60] 
console.log(queue.dequeue()); 
  
// returns 20 
console.log(queue.front()); 
  
// removes 20 
// queue contains [30, 40, 50, 60] 
console.log(queue.dequeue()); 
  
// printing the elements of the queue 
// prints [30, 40, 50, 60] 
console.log(queue.printQueue()); 

File: Data_Structures/Queue/queue.cc
#include<iostream>

class queue {
private:
  static const int MAXN = 1000000;
  int first;
  int last;
  int D[MAXN];
  
public:
  queue() {
    first = 0;
    last = 0;
  };
  ~queue(){};
  
  void push(int x) {
    D[last] = x;
    last = (last + 1)%MAXN;
  }
  
  void pop() {
    first = (first + 1)%MAXN;
  }
  
  int front() {
    return D[first];
  }
};

int main() {
  queue Q;
  int t, x;
  while (std::cin >> t) {
    if (t) {
      std::cin >> x;
      Q.push(x);
    }
    else {
      std::cout << Q.front() << '\n';
      Q.pop();
    }
  }
}
